riscv-page-tables/src/page_table.rs

```diff
@@ -658,16 +665,21 @@ impl<T: FirstStagePagingMode> FirstStagePageTable<T> {
         num_pages: u64,
         get_pte_page: &mut dyn FnMut() -> Option<Page<InternalClean>>,
     ) -> Result<FirstStageMapper<T>> {
-        if page_size.is_huge() {
-            return Err(Error::PageSizeNotSupported(page_size));
+        let mut inner = self.inner.lock();
+
+        if !addr.is_aligned(page_size) {
+            return Err(Error::AddressMisaligned(addr.bits()));
         }
```

Comment:
In general it's better to put these sanity checks first, before acquiring a lock, if possible.

Reply:
not done